src/rvpt/window.cpp

```diff
@@ -14,10 +14,12 @@ Window::Window(Window::Settings settings) : active_settings(settings)
 
     glfwWindowHint(GLFW_CLIENT_API, GLFW_NO_API);
     glfwWindowHint(GLFW_RESIZABLE, GLFW_FALSE);
-    window_ptr = glfwCreateWindow(settings.width, settings.height,
-                                  settings.title, NULL, NULL);
-    if (window_ptr == nullptr)
-        std::cerr << "Failed to create a glfw window" << '\n';
+    window_ptr = glfwCreateWindow(settings.width, settings.height, settings.title, nullptr, nullptr);
+    if (window_ptr == nullptr) std::cerr << "Failed to create a glfw window" << '\n';
```

Comment:
If it errors out, we need to return immediately, calling the subsequent functions with a nullptr for window_ptr will lead to bad stuff happening.

Reply:
This is now handled and no glfw functions are called if window creation fails

Reply:
Looks good. 
Purely from a style perspective, I prefer a 'early out' instead of this `if (fail) else (succeed)` but thats a nitpick for something thats functionally equivalent.